examples/vision/integrated_gradients.py

```diff
@@ -0,0 +1,474 @@
+
+        # 4. Only keep values that are bigger than the lower_end
+        transformed_attributions *= transformed_attributions >= lower_end
+
+        # 5. Clip values and return
+        transformed_attributions = np.clip(transformed_attributions, 0.0, 1.0)
+        return transformed_attributions
+
+    def get_thresholded_attributions(self, attributions, percentage):
+        if percentage == 100.0:
+            return np.min(attributions)
+
+        # 1. Flatten the attributions
+        flatten_attr = attributions.flatten()
+
+        # 2. Get the sum of the attributions
+        total = np.sum(flatten_attr)
+
+        # 3. Sort the attributions from largest to smallest.
+        sorted_attributions = np.sort(np.abs(flatten_attr))[::-1]
+
+        # 4. Calculate the percentage of the total sum that each attribution
+        # and the values about it contribute.
+        cum_sum = 100.0 * np.cumsum(sorted_attributions) / total
+
+        # 5. threshold the attributions by the percentage
+        indices_to_consider = np.where(cum_sum >= percentage)[0][0]
+
+        # 6. select the desired attributions and return
+        attributions = sorted_attributions[indices_to_consider]
+        return attributions
+
+    def binarize(self, attributions, threshold=0.001):
+        return attributions > threshold
+
+    def morphological_cleanup_fn(self, attributions, structure=np.ones((4, 4))):
+        closed = ndimage.grey_closing(attributions, structure=structure)
+        opened = ndimage.grey_opening(closed, structure=structure)
+        return opened
+
+    def draw_outlines(
+        self, attributions, percentage=90, connected_component_structure=np.ones((3, 3))
+    ):
+        # 1. Binarize the attributions.
+        attributions = self.binarize(attributions)
+
+        # 2. fill the gaps
+        attributions = ndimage.binary_fill_holes(attributions)
+
+        # 3. Compute connected components
+        connected_components, num_comp = ndimage.measurements.label(
+            attributions, structure=connected_component_structure
+        )
+
+        # 4. Go sum up the attributions for each component
+        total = np.sum(attributions[connected_components > 0])
+        component_sums = []
+
+        for comp in range(1, num_comp + 1):
+            mask = connected_components == comp
+            component_sum = np.sum(attributions[mask])
+            component_sums.append((component_sum, mask))
+
+        # 5. Compute the percentage of top components to keep.
+        sorted_sums_and_masks = sorted(component_sums, key=lambda x: x[0], reverse=True)
+        sorted_sums = list(zip(*sorted_sums_and_masks))[0]
+        cumulative_sorted_sums = np.cumsum(sorted_sums)
+        cutoff_threshold = percentage * total / 100
+        cutoff_idx = np.where(cumulative_sorted_sums >= cutoff_threshold)[0][0]
+
+        if cutoff_idx > 2:
+            cutoff_idx = 2
+
+        # 6.Set the values for the kept components.
```

Comment:
The comments are good. Make sure to proofread for inconsistencies and typos (e.g. lack of space after period above, double period on line 363, use of comma instead of period, etc)